pkg/indexerpoc/README.md

```diff
@@ -0,0 +1,203 @@
+# Indexer
+
+Usage example:
+
+```go
+package main
+
+import (
+    "context"
+    "time"
+
+    "github.com/ethereum/go-ethereum/core/types"
+    "github.com/xmtp/xmtpd/pkg/indexerpoc"
+    "github.com/xmtp/xmtpd/pkg/indexerpoc/contract"
+    "github.com/xmtp/xmtpd/pkg/indexerpoc/contract/group_message_broadcaster"
+    "github.com/xmtp/xmtpd/pkg/indexerpoc/contract/identity_update_broadcaster"
+    "go.uber.org/zap"
+)
+
+func main() {
+
+ctx := context.Background()
+
+storage := indexerpoc.NewMemoryStorage()
+log, _ := zap.NewProduction()
+defer log.Sync()
+
+// Define networks.
+ethereumNetwork := indexerpoc.NewNetwork(
+    "ethereum",
+    1,
+    "https://mainnet.infura.io/v3/YOUR_API_KEY",
+    12 * time.Second,
+)
+
+arbitrumNetwork := indexerpoc.NewNetwork(
+    "arbitrum",
+    42161,
+    "https://arb1.arbitrum.io/rpc",
+    250 * time.Millisecond,
+)
+
+// Create indexer.
+manager, err := indexerpoc.NewIndexer(
```

Comment:
if we are writing things from scratch I would prefer to use the golang [Option Struct Patter](https://medium.com/@omid.jn/the-option-pattern-in-go-9edf55e88320). I've been meaning to introduce it in a few other places, but since we are starting from scratch here, it would be the perfect candidate.